aspnetcore/mvc/controllers/filters.md

```diff
@@ -1,124 +1,167 @@
 
-Custom filters can be created to handle cross-cutting concerns. Filters can avoid duplicating code across actions. For example, an error handling exception filter could consolidate error handling.
+This document applies to Razor Pages, API controllers, and controllers with views.
 
 [View or download sample from GitHub](https://github.com/aspnet/AspNetCore.Docs/tree/master/aspnetcore/mvc/controllers/filters/sample).
 
 ## How filters work
 
-Filters run within the *MVC action invocation pipeline*, sometimes referred to as the *filter pipeline*.  The filter pipeline runs after MVC selects the action to execute.
+Filters run within the *ASP.NET Core action invocation pipeline*, sometimes referred to as the *filter pipeline*.  The filter pipeline runs after ASP.NET Core selects the action to execute.
 
-![The request is processed through Other Middleware, Routing Middleware, Action Selection, and the MVC Action Invocation Pipeline. The request processing continues back through Action Selection, Routing Middleware, and various Other Middleware before becoming a response sent to the client.](filters/_static/filter-pipeline-1.png)
+![The request is processed through Other Middleware, Routing Middleware, Action Selection, and the ASP.NET Core Action Invocation Pipeline. The request processing continues back through Action Selection, Routing Middleware, and various Other Middleware before becoming a response sent to the client.](filters/_static/filter-pipeline-1.png)
 
 ### Filter types
 
-Each filter type is executed at a different stage in the filter pipeline.
+Each filter type is executed at a different stage in the filter pipeline:
 
-* [Authorization filters](#authorization-filters) run first and are used to determine whether the current user is authorized for the current request. They can short-circuit the pipeline if a request is unauthorized. 
+* [Authorization filters](#authorization-filters) run first and are used to determine whether the user is authorized for the request. Authorization filters short-circuits the pipeline if the request is unauthorized.
 
-* [Resource filters](#resource-filters) are the first to handle a request after authorization.  They can run code before the rest of the filter pipeline, and after the rest of the pipeline has completed. They're useful to implement caching or otherwise short-circuit the filter pipeline for performance reasons. They run before model binding, so they can influence model binding.
+* [Resource filters](#resource-filters) run after authorization.  They can run code before the rest of the filter pipeline, and after the rest of the pipeline has completed. They're useful to implement caching or otherwise short-circuit the filter pipeline for performance reasons. They run before (and after) model binding, so they can influence model binding.
```

Comment:
Resource Filters run before model-binding (OnResourceExecuting), but there's nothing special about them running after (OnResourceExecuted) that applies to model-binding. I can only speculate, but I think it's specifically called out that they run before model-binding because they're the last filter in the pipeline that does. Action Filters are next to run, but do so after model-binding. Technically, all the filters run after model-binding because they have an XXXExecuted, but I don't think that's what was meant here.